portfolio/src/main/webapp/script.js

```diff
@@ -48,14 +48,105 @@ function showMoreOrLessProjects() {
 }
 
 function getAndPrintComments() {
+
+    /*fetch('/data').then((response) => response.json()).then((comments) => {
     
-    fetch('/data').then((response) => response.json()).then((message) => {
-        var comments = "";
-        for(var i=0; i<message.length; i++){
-            comments+= "<br>"+message[i]+"<hr>";
-            console.log(message[i]);
-        } 
-        document.getElementById('comments-container').innerHTML = "<br><p style='font-size: larger;'><strong><em> Comments: </em></strong></p>"+ comments;
-    });
-    
+    const commentListElement = document.getElementById('comments-container');
+    const buttonWithDropdown= "<div class='dropdown'>"+
+                    "<p><button class='commentSettings' onclick='commentSettings()'><i></i>&#9881;</button></p>"+
+                    "<div id='myDropdown' class='dropdown-content'>"+
+                        "<a href='#' onclick='deleteAllComments()'>Delete all comments</a>"+
+                        "<a href='#'>Set max no. of comments</a>"+
+                    "</div>"+
+                "</div>";
+    var commentsToBeAdded ="";
+    for(const comment of comments) {
+        commentsToBeAdded+= "<br>"+comment;
+    }
+    commentListElement.innerHTML= "<br><p style='font-size: larger;'><strong><em> Comments: </em></strong></p>"+ commentsToBeAdded+ buttonWithDropdown;
+  });*/
+  const ids = new Array();
+  const commentListElement = document.getElementById('comments-container');
+  fetch('/data').then(response => response.json()).then((comments) => {
+    comments.forEach((comment) => {
+      commentListElement.appendChild(createCommentElement(comment));
+      ids.push(comment.id);
+    })
+  });
+
+  const commentsDivElement = document.getElementById('comments-div');
+  //Creating a delete all button
```

Comment:
Why don't you define delete-all button in html. You can programmatically disable it when there are no buttons to display.

Same for the settings button

Reply:
Nice idea, will do that.

Reply:
At first, I replaced all the text from my comments-container so I tried to keep it all in script. But now I guess I no longer need to do that.